### ISSUE TITLE
`storage.defineItem` throws in content scripts

### ISSUE THREAD

**Issue by hikiko4ern** [wxt-dev/wxt#485]
### Describe the bug

In #478, `browser.runtime.onInstalled.addListener` was added to `storage.defineItem`, which throws an error in content scripts
```
TypeError: browser$1.runtime.onInstalled is undefined storage.js:272:6
```
at this line
https://github.com/wxt-dev/wxt/blob/3f149ac4fafbc7a00a285a8f3eb6f3bef61df10b/src/storage.ts#L298

### To Reproduce

[wxt-storage.zip](https://github.com/wxt-dev/wxt/files/14391735/wxt-storage.zip)

1. Install dependencies: `pnpm i`
2. Start dev mode: `pnpm dev`
3. Load the extension and open any page (e.g. https://www.google.com)
4. An error will be logged in the console of this page, `main` of the content script will not execute

### Expected behavior

It shouldn't throw an error in the content scripts 🙃

### Environment

```
System:
  OS: Linux 5.15 Debian GNU/Linux trixie/sid
  CPU: (32) x64 AMD Ryzen 9 5950X 16-Core Processor
  Memory: 66.36 GB / 70.67 GB
  Container: Yes
  Shell: 5.9 - /bin/zsh
Binaries:
  Node: 20.11.0 - /mnt/wslg/runtime-dir/fnm_multishells/11328_1708658885127/bin/node
  Yarn: 4.0.2 - /mnt/wslg/runtime-dir/fnm_multishells/11328_1708658885127/bin/yarn
  npm: 10.2.4 - /mnt/wslg/runtime-dir/fnm_multishells/11328_1708658885127/bin/npm
  pnpm: 8.15.3 - /mnt/wslg/runtime-dir/fnm_multishells/11328_1708658885127/bin/pnpm
  bun: 1.0.29 - ~/.bun/bin/bun
Browsers:
  Chrome: 120.0.6099.216
  Chromium: 120.0.6099.199
npmPackages:
  wxt: ^0.17.2 => 0.17.2
```

### Additional context

#478 also causes another problem: because the migrations are now applied asynchronously, the code starts executing with the past version's data.

So, for example, if migration adds a new field to the structure, when the code is executed, this field will not be in the storage, and operations with it will fail at runtime (although everything will be fine by types).
Maybe I should create a separate issue for this (there is a repro at https://github.com/hikiko4ern/wxt-storage-migrations).

**Comment by aklinker1** [comment]
Fixed main issue in https://github.com/wxt-dev/wxt/commit/77e6d1f5a4d30acf85eeded0348f5ea89f1fdf24, will fix the migration issues as well before closing this PR, no need for a second issue.

**Comment by aklinker1** [comment]
Fixed in [`v0.17.3`](https://github.com/wxt-dev/wxt/releases/tag/v0.17.3).
